qiskit_experiments/library/randomized_benchmarking/data/generate_clifford_data.py

```diff
@@ -0,0 +1,187 @@
+    Instead, only the Cliffords that consist of a single gate defined in ``_CLIFF_SINGLE_GATE_MAP_2Q``
+    are considered as the target Clifford. That means the compose table of
+    all-Cliffords by single-gate-cliffords is created.
+    It is sufficient because every Clifford on the right hand side of Clifford composition
+    can be broken down into a sequence of single gate Cliffords.
+    This greatly reduces the storage space for the array of
+    composition results (from O(n^2) to O(n)), where n is the number of Cliffords.
+    """
+    products = scipy.sparse.lil_matrix((NUM_CLIFFORD_2Q, NUM_CLIFFORD_2Q), dtype=int)
+    for lhs in range(NUM_CLIFFORD_2Q):
+        for rhs in _CLIFF_SINGLE_GATE_MAP_2Q.values():
+            composed = _TO_CLIFF_2Q[lhs].compose(_TO_CLIFF_2Q[rhs])
+            products[lhs, rhs] = _TO_INT_2Q[_hash_cliff(composed)]
+    return products.tocsr()
+
+
+_GATE_LIST_1Q = [
+    IGate(),
+    HGate(),
+    SXdgGate(),
+    SGate(),
+    XGate(),
+    SXGate(),
+    YGate(),
+    ZGate(),
+    SdgGate(),
+]
+
+
+def gen_cliff_single_1q_gate_map():
+    """
+    Generates a dict mapping numbers to 1Q Cliffords, which is set to ``_CLIFF_SINGLE_GATE_MAP_1Q``
+    in :mod:`~qiskit_experiment.library.randomized_benchmarking.clifford_utils`.
+    Based on it, we build a mapping from every single-gate-clifford to its number.
+    The mapping actually looks like {(gate, (0, )): num}.
+    """
+    table = {}
+    for gate in _GATE_LIST_1Q:
+        qc = QuantumCircuit(1)
+        qc.append(gate, [0])
+        num = _TO_INT_1Q[_hash_cliff(Clifford(qc))]
+        table[(gate.name, (0,))] = num
+
+    return table
+
+
+def gen_cliff_single_2q_gate_map():
+    """
+    Generates a dict mapping numbers to 2Q Cliffords, which is set to ``_CLIFF_SINGLE_GATE_MAP_2Q``
+    in :mod:`~qiskit_experiment.library.randomized_benchmarking.clifford_utils`.
+    Based on it, we build a mapping from every single-gate-clifford to its number.
+    The mapping actually looks like {(gate, (0, 1)): num}.
+    """
+    gate_list_2q = [
+        CXGate(),
+        CZGate(),
+    ]
+    table = {}
+    for gate, qubit in itertools.product(_GATE_LIST_1Q, [0, 1]):
+        qc = QuantumCircuit(2)
+        qc.append(gate, [qubit])
+        num = _TO_INT_2Q[_hash_cliff(Clifford(qc))]
+        table[(gate.name, (qubit,))] = num
+
+    for gate, qubits in itertools.product(gate_list_2q, [(0, 1), (1, 0)]):
+        qc = QuantumCircuit(2)
+        qc.append(gate, qubits)
+        num = _TO_INT_2Q[_hash_cliff(Clifford(qc))]
+        table[(gate.name, qubits)] = num
+
+    return table
+
+
+if __name__ == "__main__":
+    if _CLIFF_SINGLE_GATE_MAP_1Q != gen_cliff_single_1q_gate_map():
+        raise Exception(
+            "_CLIFF_SINGLE_GATE_MAP_1Q must be generated by gen_cliff_single_1q_gate_map()"
+        )
+    np.savez_compressed("clifford_inverse_1q.npz", table=gen_clifford_inverse_1q())
```

Comment:
I don't think this code depends on terra because the semantics of gates won't be changed. Also CliffordUtils (but not this code) defines the num<->Clifford mapping. Updates in CliffordUtils are tested by unittest and updates in this code are tested by assertions embedded in the code. The person who updates CliffordUtils is responsible to let users know the change in num<->Clifford mapping if they do so.